### DISCUSSION TITLE
Bevy 0.9 fails to compile on rust stable on MacOS

### DISCUSSION THREAD

**Discussion by promgamer** [bevyengine/bevy#6617] [Q&A]
I have a simple project that just imports bevy 0.9 and tries to compile a hello-world project

```
error[E0554]: `#![feature]` may not be used on the stable release channel
   --> /Users/.cargo/registry/src/github.com-1ecc6299db9ec823/thiserror-1.0.37/src/lib.rs:238:34
    |
238 | #![cfg_attr(provide_any, feature(provide_any))]
```

my setup:

```
stable-x86_64-apple-darwin (default)
rustc 1.65.0 (897e37553 2022-11-02)
```

Am I the only one having this issue?

**Comment by nicopap** [comment]
Are you using a cargo workspace? If so, you may need to add a `resolver = "2"` line to your workspace's `Cargo.toml`

**Reply by promgamer** [reply]
No cargo workspace. My `Cargo.toml` is as simple as 

```
[package]
name = "test"
version = "0.1.0"

[dependencies]
bevy = "0.9.0"
```

**Reply by nicopap** [reply]
Then try running `rustup update`, `cargo update`, then try to build again. I'm not sure this fixes it, but you never know.

**Reply by promgamer** [reply]
No, did not work 😞 - I've also tried to remove `Cargo.lock` and start from scratch

**Reply by nicopap** [reply]
Found this, says running `cargo clean` fixes it

* https://github.com/dtolnay/thiserror/issues/192

**Reply by tim-blackbird** [reply]
You should add `edition = "2021"` under `[package]`

**Reply by promgamer** [reply]
It does fix it. Strange I've searched that repo earlier today and did not manage to find that one 😓 

@devil-ira noted ✅
